### PR TITLE
added a null check on dtb when pressing yes on dialog

### PR DESCRIPTION
**Pull request by achintya-7**
prevets null value exception by checking if `dtb` is null or not
references the issue #89

### CODE REVIEW

File: lib/widgets/buildTasks.dart [hunk 1]
```diff
@@ -62,161 +62,176 @@ class _TasksBuilderState extends State<TasksBuilder> {
...
+                            context: context,
+                            builder: (context) {
+                              return AlertDialog(
+                                title:
+                                    const Text('Do you want to save changes?'),
+                                actions: [
+                                  TextButton(
+                                    onPressed: () {
+                                      setStatus('completed', task.uuid);
+                                      final flutterLocalNotificationsPlugin =
```
Comment (achintya-7) [inline lib/widgets/buildTasks.dart:81]: The first null value check is here

File: lib/widgets/buildTasks.dart [hunk 2]
```diff
@@ -62,161 +62,176 @@ class _TasksBuilderState extends State<TasksBuilder> {
...
+                                    const Text('Do you want to save changes?'),
+                                actions: [
+                                  TextButton(
+                                    onPressed: () {
+                                      setStatus('deleted', task.uuid);
+                                      final flutterLocalNotificationsPlugin =
+                                          FlutterLocalNotificationsPlugin();
+                                      DateTime? dtb = task.due;
+
+                                      if (dtb == null) {
```
Comment (achintya-7) [inline lib/widgets/buildTasks.dart:149]: The 2nd one for the delete operation is here.

File: lib/widgets/buildTasks.dart [hunk 3]
```diff
@@ -62,161 +62,176 @@ class _TasksBuilderState extends State<TasksBuilder> {
...
+                                title:
+                                    const Text('Do you want to save changes?'),
+                                actions: [
+                                  TextButton(
+                                    onPressed: () {
+                                      setStatus('completed', task.uuid);
+                                      final flutterLocalNotificationsPlugin =
+                                          FlutterLocalNotificationsPlugin();
+                                      DateTime? dtb = task.due;
+
```
Comment (achintya-7) [inline lib/widgets/buildTasks.dart:84]: The first null check for complete dialog is here

File: lib/widgets/buildTasks.dart [hunk 4]
```diff
@@ -62,161 +62,176 @@ class _TasksBuilderState extends State<TasksBuilder> {
...
+                                title:
+                                    const Text('Do you want to save changes?'),
+                                actions: [
+                                  TextButton(
+                                    onPressed: () {
+                                      setStatus('deleted', task.uuid);
+                                      final flutterLocalNotificationsPlugin =
+                                          FlutterLocalNotificationsPlugin();
+                                      DateTime? dtb = task.due;
+
```
Comment (achintya-7) [inline lib/widgets/buildTasks.dart:148]: The 2nd null check for the delete operation is here.

File: lib/widgets/buildTasks.dart [hunk 5]
```diff
@@ -62,161 +62,176 @@ class _TasksBuilderState extends State<TasksBuilder> {
...
+                                      setStatus('completed', task.uuid);
+                                      final flutterLocalNotificationsPlugin =
+                                          FlutterLocalNotificationsPlugin();
+                                      DateTime? dtb = task.due;
+
+                                      if (dtb == null) {
+                                        flutterLocalNotificationsPlugin
+                                            .cancel(task.uuid.hashCode);
+                                        Navigator.of(context).pop();
+                                        return;
```
Comment (achintya-7) [inline lib/widgets/buildTasks.dart:89]: The first null check for complete dialog is here

File: lib/widgets/buildTasks.dart [hunk 6]
```diff
@@ -62,161 +62,176 @@ class _TasksBuilderState extends State<TasksBuilder> {
...
+                                      setStatus('deleted', task.uuid);
+                                      final flutterLocalNotificationsPlugin =
+                                          FlutterLocalNotificationsPlugin();
+                                      DateTime? dtb = task.due;
+
+                                      if (dtb == null) {
+                                        flutterLocalNotificationsPlugin
+                                            .cancel(task.uuid.hashCode);
+                                        Navigator.of(context).pop();
+                                        return;
```
Comment (achintya-7) [inline lib/widgets/buildTasks.dart:153]: The 2nd null check for the delete operation is here.

**Comment by NishantSinghal19** [comment]
No we can get null value in due

**Comment by NishantSinghal19** [comment]
And i think this is not needed so I am closing this pr

**Pull request closed without merging**